### DISCUSSION TITLE
Multiple PocketBase applications on a single server

### DISCUSSION THREAD

**Discussion by breakingbad45** [pocketbase/pocketbase#3774] [Q&A]
i am trying to host multiple poocketbase in a single droplet 
First One -
`/root/pb/pocketbase serve yourdomain1.com`
Second One -
`/root/pb2/pocketbase serve yourdomain2.com`

First one successfully executed but when i trying to execute second one its showing an error
`listen tcp 0.0.0.0:443: bind: address already in use`
How to fix this?

**Comment by ganigeorgiev** [comment]
Both instances will try to bind to the same ports (80 and 443) and this is not allowed (_unless `SO_REUSEPORT`/`SO_REUSEADDR` is used but I don't really have much experience with directly using these socket options in production environments and I don't know how well that will work_).

If you need to host multiple applications on the same server, my general recommendation is to put them behind a reverse proxy like nginx, apache, caddy, etc. (_you may also need [certbot](https://certbot.eff.org/) or similar to take care for the certificate_).
This way only the reverse proxy will bind to the 80/443 ports and it will proxy the requests to the other applications that could bind to some local nonprivileged ports.

You can start each executable with a specific port using the `--http` flag:

```sh
/root/pb/pocketbase serve --http='127.0.0.1:8091'
```
Note that the above port will have to mapped in your vhost reverse proxy configuration.

A minimal nginx example config could be found in https://pocketbase.io/docs/going-to-production/#using-reverse-proxy.

**Comment by deniskipeles** [comment]
> i am trying to host multiple poocketbase in a single droplet First One - `/root/pb/pocketbase serve yourdomain1.com` Second One - `/root/pb2/pocketbase serve yourdomain2.com`
> 
> First one successfully executed but when i trying to execute second one its showing an error `listen tcp 0.0.0.0:443: bind: address already in use` How to fix this?

## You should try [Caprover](https://caprover.com) if you are more of a GUI guy. Or [Dokku](https://dokku.com) if you are familiar with heroku cli
